src/models/feature.cpp

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+#include "feature.h"
+#include "settingsholder.h"
+#include "../translations/generated/l18nstrings.h"
+#include "logger.h"
+#include "update/versionapi.h"
+
+#include <QScopeGuard>
+#include <QList>
+
+namespace {
+  Logger logger(LOG_MODEL, "Feature");
+  QMap<QString,Feature*> all_features = QMap<QString,Feature*>();
+}
+
+Feature::Feature(QString id, L18nStrings::String name_id, bool isMajor,
+                 L18nStrings::String shortDesc_id, L18nStrings::String desc_id,
+                 QString imgPath, QString iconPath, QString aReleaseVersion,
+                 bool writeable)
+    : QObject(qApp),
+      m_id(id),
+      m_name_id(name_id),
+      m_majorFeature(isMajor),
+      m_shortDescription_id(shortDesc_id),
+      m_description_id(desc_id),
+      m_imagePath(imgPath),
+      m_iconPath(iconPath),
+      m_devModeWriteable(writeable) {
+  if (m_devModeWriteable) {
+    connect(SettingsHolder::instance(),
+            &SettingsHolder::devModeFeatureFlagsChanged, this,
+            [this]() { emit supportedChanged(); });
+  }
+  all_features.insert(m_id, this);
+
+  auto releaseVersion = VersionApi::stripMinor(aReleaseVersion);
+  auto currentVersion = VersionApi::stripMinor(APP_VERSION);
+
+  auto cmp = VersionApi::compareVersions(releaseVersion, currentVersion);
+  if (cmp == -1) {
+    // Release verion < currVersion
+    m_released = true;
+    m_new = false;
+  } else if (cmp == 0) {
+    // Release verion === currVersion
+    m_released = true;
+    m_new = true;
+  } else {
+    // releaseVersion > currVersion
+    m_released = false;
+    m_new = false;
+  }
+}
+
+// Static
+QList<Feature*> Feature::getAll(){
+    return all_features.values();
+}
+// Static
+const Feature* Feature::get(const QString& featureID){
+    if(!all_features.contains(featureID)){
+        // Todo: CRASH
+        return nullptr;
+    };
+    return all_features[featureID];
+}
+
+bool Feature::isDevModeEnabled() const{
+    if(!m_devModeWriteable){
+        return false;
+    }
+    const auto settings = SettingsHolder::instance();
+    return settings->hasDevModeFeatureFlag(m_id);
+}
+
+bool Feature::isSupported() const{
+    logger.debug() << "Check feature " << m_id;
+    if(isDevModeEnabled()){
+        logger.debug() << "Devmode Enabled " << m_id;
+        return true;
+    }
+    if(!m_released){
+        logger.debug() << "Not released " << m_id;
+        return false;
+    }
+    logger.debug() << "Check Support" << m_id;
+    return checkSupportCallback();
+}
+
+bool Feature::isSeen() const {
```

Comment:
Maybe we should keep this out of this pr.
What about if we land your pr without this, and we let the Whatsnew implement it. I like features as a separate thing.